Cannot call methods on map literals

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** 0.2.2 02f0a30
**OS:** Arch Linux

**What did you do?**
`v -prod test.v`
```v
fn main() {
	a := 1 in map{
	1: 1
}.keys().map(map{
	1: 1
}[it])
	println(a)
}
```

**What did you expect to see?**
```v
true
```
 
**What did you see instead?**
```
/tmp/v/test.7091847091740794489.tmp.c:10144:39: error: lvalue required as unary ‘&’ operand
10144 |         Array_int _t1_orig = map_keys(&new_map_init(&map_hash_int_4, &map_eq_int_4, &map_clone_int_4, &map_free_nop, 1, sizeof(int), sizeof(int), _MOV((int[1]){1, }), _MOV((int[1]){1, })));
      |                                       ^
...
```

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -778,7 +778,13 @@ fn (mut g Gen) method_call(node ast.CallExpr) {
 			&& node.name in ['first', 'last', 'repeat'] {
 			g.write('*')
 		}
-		g.expr(node.left)
+		if node.left is ast.MapInit {
+			g.write('(map[]){')
+			g.expr(node.left)
+			g.write('}[0]')
+		} else {
+			g.expr(node.left)
+		}
 		if node.from_embed_type != 0 {
 			embed_name := typ_sym.embed_name()
 			if node.left_type.is_ptr() {

--- a/vlib/v/tests/map_literals_method_call_test.v
+++ b/vlib/v/tests/map_literals_method_call_test.v
@@ -1,0 +1,10 @@
+fn test_map_literals_method_call() {
+	a := 1 in map{
+		1: 1
+	}.keys().map(map{
+		1: 1
+	}[it])
+
+	println(a)
+	assert a
+}

